core/src/main/java/hudson/model/EnvironmentContributingAction.java

```diff
@@ -45,12 +50,39 @@
  */
 public interface EnvironmentContributingAction extends Action {
     /**
+     * Called by {@link Run} or {@link AbstractBuild} to allow plugins to contribute environment variables.
+     *
+     * @param run
+     *      The calling build. Never null.
+     * @param node
+     *      The node execute on. Can be null.
```

Comment:
`Can be {@code null} when the Run is not binded to the node (e.g. in Pipeline outside the {@code node() step})`